### PR TITLE
docs(api): add string api examples

### PR DESCRIPTION
**Pull request by cpcloud**
This PR adds examples to almost every user-facing string API. I did not add examples where the duckdb backend does not implement a particular API.

### CODE REVIEW

File: ibis/expr/types/strings.py [hunk 1]
```diff
@@ -241,6 +576,22 @@ def repeat(self, n: int | ir.IntegerValue) -> StringValue:
...
+        >>> t.s.repeat(5)
+        ┏━━━━━━━━━━━━━━┓
+        ┃ Repeat(s, 5) ┃
+        ┡━━━━━━━━━━━━━━┩
+        │ string       │
+        ├──────────────┤
+        │ aaaaa        │
+        │ bbbbb        │
+        │ ccccc        │
+        └──────────────┘
```
Comment (gforsyth) [inline ibis/expr/types/strings.py:594]:
```suggestion
        >>> t = ibis.memtable({"s": ["a", "bb", "c"]})
        >>> t.s.repeat(5)
        ┏━━━━━━━━━━━━━━┓
        ┃ Repeat(s, 5) ┃
        ┡━━━━━━━━━━━━━━┩
        │ string       │
        ├──────────────┤
        │ aaaaa        │
        │ bbbbbbbbbb   │
        │ ccccc        │
        └──────────────┘
```

File: ibis/expr/types/strings.py [hunk 2]
```diff
@@ -316,15 +693,23 @@ def lpad(
...
+        ┏━━━━━━━━━━━━━━━━━┓
+        ┃ LPad(s, 5, '-') ┃
+        ┡━━━━━━━━━━━━━━━━━┩
+        │ string          │
+        ├─────────────────┤
+        │ --abc           │
+        │ --def           │
+        │ --ghi           │
+        └─────────────────┘
         """
```
Comment (gforsyth) [inline ibis/expr/types/strings.py:713]:
```suggestion
        >>> t = ibis.memtable({"s": ["abc", "def", "ghij"]})
        >>> t.s.lpad(5, "-")
        ┏━━━━━━━━━━━━━━━━━┓
        ┃ LPad(s, 5, '-') ┃
        ┡━━━━━━━━━━━━━━━━━┩
        │ string          │
        ├─────────────────┤
        │ --abc           │
        │ --def           │
        │ -ghij           │
        └─────────────────┘
        """
```

File: ibis/expr/types/strings.py [hunk 3]
```diff
@@ -344,18 +729,26 @@ def rpad(
...
+        >>> t.s.rpad(5, "-")
+        ┏━━━━━━━━━━━━━━━━━┓
+        ┃ RPad(s, 5, '-') ┃
+        ┡━━━━━━━━━━━━━━━━━┩
+        │ string          │
+        ├─────────────────┤
+        │ abc--           │
+        │ def--           │
+        │ ghi--           │
+        └─────────────────┘
```
Comment (gforsyth) [inline ibis/expr/types/strings.py:751]:
```suggestion
        >>> t = ibis.memtable({"s": ["abc", "def", "ghij"]})
        >>> t.s.rpad(5, "-")
        ┏━━━━━━━━━━━━━━━━━┓
        ┃ RPad(s, 5, '-') ┃
        ┡━━━━━━━━━━━━━━━━━┩
        │ string          │
        ├─────────────────┤
        │ abc--           │
        │ def--           │
        │ ghij-           │
        └─────────────────┘
```

**Review by gforsyth — APPROVED** [review]
This is great @cpcloud !

I added in a few small suggestions but feel very free to ignore all of them and just smash merge.

**Comment by cpcloud** [comment]
I wonder if the merge queue doesn't work for things that have already passed?

**Comment by cpcloud** [comment]
I'm gonna remove it from the queue for now, seems like it's having trouble detecting things that have already passed.

**Pull request merged**